src/System.Collections/src/System/Collections/Generic/SortedSet.TreeSubSet.cs

```diff
@@ -45,12 +45,6 @@ public TreeSubSet(SortedSet<T> Underlying, T Min, T Max, bool lowerBoundActive,
                 VersionCheckImpl();
             }
 
-            private TreeSubSet(SerializationInfo info, StreamingContext context)
```

Comment:
Expanding on that a little, I'd think having these sorts of types be serializable limits our internal implementation much more than the restrictions on field names. For example, lots of the older collections have various inner collections that could probably be replaced by generics.